### ISSUE TITLE
Dense layer gets wrong dim?

### ISSUE THREAD

**Issue by abhishmitra** [malongxuan/QAselection#1]
Hi,
I seem to be getting an error here: 
`Dense_Align_q = Dense(hidden_dim,activation="tanh")(question_proj)`

Error:
`Exception: Input 0 is incompatible with layer dense_7: expected ndim=2, found ndim=3
`

Also please update the keras version. I think this runs on 1.1 not 1.0

**Comment by malongxuan** [comment]
Hi,
I've encountered this issue before. There are two possibilities for this problem, one is the alignment of dimensions in data preprocessing, the other is that every step of data processing in the program should conform to keras specifications. The code I uploaded works well in my environment. If the issue still cannot be solved, please remind me. Thank you for your attention to my work.

**Comment by abhishmitra** [comment]
Thank you for replying! 
I think the issue definitely occurs before the data is fed in.

That is, during model setup.

It would be great if you could share the exact configuration that worked for you (or could confirm if it runs with the readme.md configurations)..

**Comment by haniehkh18** [comment]
I also have the same issue with code.

**Comment by haniehkh18** [comment]
> Hi,
> I seem to be getting an error here:
> `Dense_Align_q = Dense(hidden_dim,activation="tanh")(question_proj)`
> 
> Error:
> `Exception: Input 0 is incompatible with layer dense_7: expected ndim=2, found ndim=3 `
> 
> Also please update the keras version. I think this runs on 1.1 not 1.0
Hello Abhish
@abhishmitra 
Where you able to resolve the issue?

**Comment by malongxuan** [comment]
Hi,

@haniehkh18，I'm sure that the initial experimental environment is the same as what I gave (python 2.7/Theano 0.8/keras 1.0), and the configuration of some other libraries in the experiment may need to be updated according to the situation. 

In the course of the experiment, once I encounter the problems you and @abhishmitra have mentioned, I use "print ("name. _keras_shape:", name. _keras_shape)" to track where the dimension errors occur, which is usually caused by two reasons given on Mar 14.

Since I can't continue to use the original server after graduation, I will try my best to reproduce the experimental environment with the resources at hand. If I find out which configurations need special attention, I will inform you at the first time.

**Comment by haniehkh18** [comment]
Dear Longxuan

I have attached a screenshot of where the error exactly occurs.

Regards;
Hanieh

**Comment by haniehkh18** [comment]
Dear Longxuan

I printed the shape and I found out that the input dimension is:
(None,22,300), but the expected dimension is 2.

Regards;
Hanieh

**Comment by malongxuan** [comment]
hi Hanieh,

In my experience, this situation is either caused by the environmental configuration (which I need some time to answer. Maybe you can try to adjust the expected dimension according to keras specifications.) or due to errors in data preprocessing ( In training, batchsize* (question (1), correct answer (k_i) and wrong answer (n-k_i) are required, k_i is the number of correct answers to the i-th questions; while in testing, 1* (question (1), all candidates (m_i) are required, m_i is the number of all candidates.)

BTW: I can't see your screenshot.

Regards,
Longxuan Ma
